### PR TITLE
[Secrets] Delete project secrets on project deletion

### PR DESCRIPTION
**Pull request by theSaarco**
Cleanup the k8s secret used for project secrets when a project is deleted.

### CODE REVIEW

File: tests/api/api/test_projects.py [hunk 1]
```diff
@@ -89,12 +90,23 @@ def test_get_non_existing_project(
...
+    k8s_secrets_mock.set_is_running_in_k8s_cluster(False)
     project_to_keep = "project-to-keep"
     project_to_remove = "project-to-remove"
     _create_resources_of_all_kinds(db, project_to_keep)
     _create_resources_of_all_kinds(db, project_to_remove)
+
+    secrets = {f"secret_{i}": "a secret" for i in range(5)}
+    k8s_secrets_mock.store_project_secrets(project_to_keep, secrets)
+    k8s_secrets_mock.store_project_secrets(project_to_remove, secrets)
+
```
Comment (Hedingber) [inline tests/api/api/test_projects.py:109]: This code should go inside `_create_resources_of_all_kinds`

File: tests/api/api/test_projects.py [hunk 2]
```diff
@@ -152,6 +164,9 @@ def test_delete_project_with_resources(
         == {}
     )
 
+    assert k8s_secrets_mock.get_project_secret_keys(project_to_remove) == []
+    assert k8s_secrets_mock.get_project_secret_data(project_to_keep) == secrets
```
Comment (Hedingber) [inline tests/api/api/test_projects.py:168]: This code should go inside `_assert_resources_in_project`

File: tests/api/api/test_projects.py [hunk 3]
```diff
@@ -89,12 +90,23 @@ def test_get_non_existing_project(
...
 def test_delete_project_with_resources(
-    db: Session, client: TestClient, project_member_mode: str
+    db: Session,
+    client: TestClient,
+    project_member_mode: str,
+    k8s_secrets_mock: tests.api.conftest.K8sSecretsMock,
 ):
+    # need to set this to False, otherwise impl will try to delete k8s resources, and will need many more
+    # mocks to overcome this.
+    k8s_secrets_mock.set_is_running_in_k8s_cluster(False)
```
Comment (Hedingber) [inline tests/api/api/test_projects.py:100]: I'm not sure I understand why it will require may more mocks, please explain (can be offline if easier)

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: tests/api/api/test_projects.py (hunk 3 above)
Comment (theSaarco) [reply tests/api/api/test_projects.py:100]: Let's take it offline.

**Review by Hedingber — APPROVED** [review]

**Pull request merged**